mne/io/snirf/tests/test_snirf.py

```diff
@@ -413,19 +419,33 @@ def test_snirf_fieldtrip_od():
 
 
 @requires_testing_data
-def test_snirf_kernel_hb():
-    """Test reading Kernel SNIRF files with haemoglobin data."""
-    raw = read_raw_snirf(kernel_hb, preload=True)
-
-    # Test data import
-    assert raw._data.shape == (180 * 2, 14)
-    assert raw.copy().pick("hbo")._data.shape == (180, 14)
-    assert raw.copy().pick("hbr")._data.shape == (180, 14)
+@pytest.mark.parametrize("kind", ("hb", "td"))
+def test_snirf_kernel(kind):
+    """Test reading Kernel SNIRF files with haemoglobin or TD data."""
+    fname = dict(hb=kernel_hb, td=kernel_td)[kind]
+    raw = read_raw_snirf(fname, preload=True)
+    if kind == "hb":
+        # Test data import
+        assert raw._data.shape == (180 * 2, 14)
+        hbo_data = raw.get_data("hbo")
+        hbr_data = raw.get_data("hbr")
+        assert hbo_data.shape == hbr_data.shape == (180, 14)
+        hbo_norm = np.nanmedian(np.linalg.norm(hbo_data, axis=-1))
+        hbr_norm = np.nanmedian(np.linalg.norm(hbr_data, axis=-1))
+        assert 1 < hbr_norm < hbo_norm < 3
+        n_nan = 20
+    else:
+        assert raw._data.shape == (1080, 14)
+        data = raw.get_data("fnirs_td_moments_amplitude")
+        assert data.shape == raw._data.shape
+        norm = np.nanmedian(np.linalg.norm(data, axis=-1))
+        assert 1e5 < norm < 1e6  # TODO: 429256, is this reasonable Molars!??
```

Comment:
This should be in units of M (mol). Looking at the values in the test file:
```
>>> import h5py
>>> import numpy as np
>>> x = h5py.File("td_moments.snirf", "r")
>>> list(x["nirs/data1/measurementList1"])  # no dataUnit, so code assumes M not uM!
['dataType', 'dataTypeIndex', 'dataTypeLabel', 'detectorIndex', 'sourceIndex', 'wavelengthIndex']
>>> np.array(x["nirs/data1/measurementList1/dataTypeLabel"]).item()
b'Time Domain - Moments - Amplitude'
>> np.nanmedian(np.linalg.norm(np.array(x["nirs/data1/dataTimeSeries"]), axis=0))
np.float64(429256.98709975014)
```
should the code assume that if no dataUnit is present, the units are actually uM?

Reply:
@julien-dubois-k this one is for you :point_up:

Reply:
b'Time Domain - Moments - Amplitude' so this is moments data.
The units will be:
- `unitless` for the "sum" moment (it's a photon count)
- some multiple of `s` for the "mean" moment (it's a time of flight)
- some multiple of `s**2` for the "var" moment (it's a variance of the time of flight)

The SNIRF spec doesn't require `dataUnit` to be populated ([here](https://github.com/fNIRS/snirf/blob/master/snirf_specification.md#nirsidatajmeasurementlistkdataunit). In cedalion it looks like they just treat unspecified dataUnit as unitless or a.u. ([here](https://github.com/ibs-lab/cedalion/blob/main/src/cedalion/io/snirf.py#L566)). I think vendors who generate SNIRF files should populate the `dataUnit` field always if there is a unit. Kernel didn't originally (hence the missing `dataUnit` in the current test files), but does now. I'll provide new test files shortly.

Reply:
test files shared [here](https://github.com/mne-tools/mne-python/pull/11064#discussion_r1838886429)